internal/cmd/gencli/templates.go

```diff
@@ -0,0 +1,466 @@
+	init{{ camelCase .SubActionPrefix }}Flags()
+	return c.Flags().Completions()
+}
+
+func (c *{{ camelCase .SubActionPrefix }}Command) Synopsis() string {
+	if extra := extra{{ camelCase .SubActionPrefix }}SynopsisFunc(c); extra != "" {
+		return extra
+	}
+
+	synopsisStr := "{{ lowerSpaceCase .ResourceType }}"
+	{{ if .SubActionPrefix }}
+	synopsisStr = fmt.Sprintf("%s %s", "{{ .SubActionPrefix }}-type", synopsisStr)
+	{{ end }}
+	return common.SynopsisFunc(c.Func, synopsisStr)
+}
+
+func (c *{{ camelCase .SubActionPrefix }}Command) Help() string {
+	init{{ camelCase .SubActionPrefix }}Flags()
+	
+	var helpStr string
+	helpMap := common.HelpMap("{{ lowerSpaceCase .ResourceType }}")
+
+	switch c.Func {
+	{{ if not .SkipNormalHelp }}
+	{{ range $i, $action := .StdActions }}
+	case "{{ $action }}":
+		helpStr = helpMap[c.Func]() + c.Flags().Help()
+	{{ end }}
+	{{ end }}
+	default:
+	{{ if .HasExtraHelpFunc }}
+		helpStr = c.extra{{ camelCase .SubActionPrefix }}HelpFunc(helpMap)
+	{{ else }}
+		helpStr = helpMap["base"]()
+	{{ end }}
+	}
+
+	// Keep linter from complaining if we don't actually generate code using it
+	_ = helpMap
+	return helpStr
+}
+
+var flags{{ camelCase .SubActionPrefix }}Map = map[string][]string{
+	{{ range $i, $action := .StdActions }}
+	{{ if eq $action "create" }}
+	"create": { "{{ kebabCase $input.Container }}-id", "name", "description" },
+	{{ end }}
+	{{ if eq $action "read" }}
+	"read": {"id"},
+	{{ end }}
+	{{ if eq $action "update" }}
+	"update": {"id", "name", "description" {{ if hasAction $input.VersionedActions "update" }}, "version" {{ end }} },
+	{{ end }}
+	{{ if eq $action "delete" }}
+	"delete": {"id"},
+	{{ end }}
+	{{ if eq $action "list" }}
+	"list": { "{{ kebabCase $input.Container }}-id" {{ if (eq $input.Container "Scope") }}, "recursive"{{ end }} },
+	{{ end }}
+	{{ end }}
+}
+
+func (c *{{ camelCase .SubActionPrefix }}Command) Flags() *base.FlagSets {
+	if len(flags{{ camelCase .SubActionPrefix }}Map[c.Func]) == 0 {
+		return c.FlagSet(base.FlagSetNone)
+	}
+
+	set := c.FlagSet(base.FlagSetHTTP | base.FlagSetClient | base.FlagSetOutputFormat)
+	f := set.NewFlagSet("Command Options")
+	common.PopulateCommonFlags(c.Command, f, "{{ if .SubActionPrefix }}{{ .SubActionPrefix }}-type {{ end }}{{ lowerSpaceCase .ResourceType }}", flags{{ camelCase .SubActionPrefix }}Map[c.Func])
+
+	extra{{ camelCase .SubActionPrefix }}FlagsFunc(c, set, f)
+
+	return set
+}
+
+func (c *{{ camelCase .SubActionPrefix }}Command) Run(args []string) int {
+	init{{ camelCase .SubActionPrefix }}Flags()
+
+	{{ if .HasExampleCliOutput }}
+	if os.Getenv("BOUNDARY_EXAMPLE_CLI_OUTPUT") != "" {
+		c.UI.Output(exampleOutput())
+		return 0
+	}
+	{{ end }}
+
+	switch c.Func {
+	case "":
+		return cli.RunResultHelp
+	{{ if .IsAbstractType }}
+	case "create", "update":
+		return cli.RunResultHelp
+	{{ end }}
+	}
+
+	c.plural = "{{ if .SubActionPrefix }}{{ .SubActionPrefix }}-type {{ end }}{{ lowerSpaceCase .ResourceType }}"
```

Comment:
Is this used anywhere else besides here in Run?  If not can you make it a local variable instead of a field of the struct?

Reply:
Ah, good catch. It used to be in some helper funcs but I migrated that in over time. Will make it local.

Reply:
Ah hah. I forgot -- it is indeed used in the targets funcs. I could special case it in the template but for now I'll keep it in. I'm able to remove c.existed though!